src/appleseed/renderer/kernel/rendering/progressive/progressiveframerenderer.cpp

```diff
@@ -417,13 +419,12 @@ namespace
                 Frame&                      frame,
                 SampleAccumulationBuffer&   buffer,
                 ITileCallback*              tile_callback,
-                const uint64                max_sample_count,
                 const double                max_fps,
                 IAbortSwitch&               abort_switch)
               : m_frame(frame)
               , m_buffer(buffer)
               , m_tile_callback(tile_callback)
-              , m_min_sample_count(min<uint64>(max_sample_count, 32 * 32 * 2))
+              , m_min_sample_count(min<uint64>(frame.get_crop_window().volume(), 32 * 32 * 2))
```

Comment:
This doesn't look correct as it's no longer taking into account the max number of samples/pixel.

Reply:
Since this number is the minimum number of samples required to first display the render progress, the previous number made sense whereas now if we incorporate the max number of samples per pixel we won't call to display for sizes < 32 * 32 * 2 unless we reach the final rendering state, which doesn't seem correct.

Using `volume` here is stating that for sizes < 32 * 32 * 2 we should just display the render once we have a window's worth of pixel samples, which would be a roughly 45 x 45 window, which renders nearly immediately.

Reply:
I think that the multiplication by two in `32 * 32 * 2` meant 2 spp for a 32x32 window.

With your changes, we start displaying when we either have 2 spp at resolution 32x32, or 1 spp at 45x45 and up, which seems fine.

I guess the logic could be simplified / cleaned up further but it's not urgent.